lib/rpcs/v1_4_5/get_account.rs

```diff
@@ -1,22 +1,39 @@
+use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
+use casper_types::account::AccountHash;
 use casper_types::{ProtocolVersion, PublicKey};
 
 use crate::{rpcs::common::BlockIdentifier, types::Account};
 
 pub(crate) const GET_ACCOUNT_METHOD: &str = "state_get_account_info";
 
+/// Identifier of an account.
+#[derive(Serialize, Deserialize, Debug, Clone, JsonSchema)]
+#[serde(deny_unknown_fields, untagged)]
+pub enum AccountIdentifier {
+    /// The public key of an account
+    PublicKey(PublicKey),
+    /// The account hash of an account
+    AccountHash(AccountHash),
+}
+
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(deny_unknown_fields)]
 pub(crate) struct GetAccountParams {
-    public_key: PublicKey,
+    ///The identifier of an Account. (named public key to match the JSON-RPC API)
```

Comment:
Hi, it seems you switch from something like public to account mostly because you are adding AccountHash. An then the api is not following so personally I would have made `pub enum PublicKeyIdentifier` because behind a key or an account hash are identifiers of a public (key whatever its form) identifier. Account is to me something else that derives from the public key or account hash but nalming things is always very subjective.

and so 
```rust
 pub(crate) fn new(
        public_key: PublicKeyIdentifier,
        
        ....
        
         GetAccountParams {
            public_key,
            block_identifier,
        }`

Reply:
or maybe 

```rust

 pub(crate) fn new(
        public_key_identifier: PublicKeyIdentifier,
        
        ....
        
         GetAccountParams {
            public_key_identifier,,
            block_identifier,
        }`
```

because this does not read very well

``` 
 "params": {
    "public_key": "account-hash-c029c14904b870e64c1d443d428c606740e82f341bea0f8542ca6494cef1383e",
    "block_identifier": null
  },
  
  But if you can break the API
  
  ``` 
 "params": {
    "public_key_identifier": "account-hash-c029c14904b870e64c1d443d428c606740e82f341bea0f8542ca6494cef1383e",
    "block_identifier": null
  },
  
  ```
  
  or indeed account_identifier
  
    ``` 
 "params": {
    "account_identifier": "account-hash-c029c14904b870e64c1d443d428c606740e82f341bea0f8542ca6494cef1383e",
    "block_identifier": null
  },

Reply:
The reason for naming it `AccountIdentifier` is that a public key is, in the nodes eyes, really an identifier of an account on the network.

That is to say an account has a set of identifiers that includes an account hash, and a public key/private key pair.

Calling it a `PublicKeyIdentifier` seems inaccurate as the entity we are asking the node to retrieve isn't a public key, rather it's all of the details related to an account associated with either a public key or an account hash.

Reply:
To me an account has a public ( any key form underneath) identifier that can take two forms right now, a public key.
Anyway it seems to me introducing the line `public_key: account_identifier,` is somehow a consequence of having to modify an arg and not adding an arg "account_identifier" to the structs and somehow deprecate "public_key" arg and also the api answer should also be answering the same way.

```
    "account": {
      "account_identifier": "account-hash-c029c14904b870e64c1d443d428c606740e82f341bea0f8542ca6494cef1383e", // or 0x key
```

or both

```
      "account_hash": "account-hash-c029c14904b870e64c1d443d428c606740e82f341bea0f8542ca6494cef1383e",
      "public_key": "0x...",

Reply:
If we apply the serde alias as you suggest [here](https://github.com/casper-network/casper-node/pull/4218#discussion_r1297674858) @zacshowa, we should be able to define this field in the client struct as `account_identifier: AccountIdentifier` without even applying the alias.